File: Command/CommandTextEditor/CommandTextEditor/TextEditor.cs
namespace CommandTextEditor
{
    public class TextEditor // Receiver
    {
        private string _text = "";

        public void InsertText(string text)
        {
            _text += text;
            Console.WriteLine($"Text:'{_text}'");
        }

        public void RemoveText(int count)
        {
            if (count > _text.Length)
            {
                count = _text.Length;
            }
            _text = _text.Substring(0, _text.Length - count);
            Console.WriteLine($"Text:'{_text}'");
        }

        public string GetText() => _text;
    }
}

File: Command/JavaExample/CommandTextEditor/src/TextEditor.java
public class TextEditor {
    private String text = "";

    public void insertText(String newText) {
        text += newText;
        System.out.println("Text: '" + text + "'");
    }

    public void removeText(int count) {
        if (count > text.length()) {
            count = text.length();
        }
        text = text.substring(0, text.length() - count);
        System.out.println("Text: '" + text + "'");
    }

    public String getText() {
        return text;
    }
}